### ISSUE TITLE
liveevent scheduling loop when updating series metadata

### ISSUE THREAD

**Issue by reiferschris** [opencast/opencast#2504]
**Describe the bug**
A clear and concise description of what you see:
*IUpdating series metadata (eg. title) in a series with a live event scheduled will cause the liveevent(s) to be republished over and over*

**To Reproduce**
1. Go to AdminUI and create a new series
2. Add  a live event in the series
3. Update series metadata (eg. title)
4. Go to events publication tab and watch live publications being added / see logs in admin and engage
5. Be prepared to stop the opencast service, deleting the event in the adminUI doesn't break the loop (logs for that are attached separately)

**Environment Information**
 - OS: [CentOS 7, CentOS 8.3]
 - Browser [Firefox]
 - Opencast version with minor release [OC 8, OC 9]
 - Capture Agents [pyCA, Extron SMP]
[opencast-2021-04-16-liveschedule.log](https://github.com/opencast/opencast/files/6323312/opencast-2021-04-16-liveschedule.log)
[opencast-2021-04-16-liveschedule-engage-deleteevent.log](https://github.com/opencast/opencast/files/6323313/opencast-2021-04-16-liveschedule-engage-deleteevent.log)
[opencast-2021-04-16-liveschedule-deleteevent.log](https://github.com/opencast/opencast/files/6323314/opencast-2021-04-16-liveschedule-deleteevent.log)
[opencast-2021-04-16-liveschedule-engage.log](https://github.com/opencast/opencast/files/6323315/opencast-2021-04-16-liveschedule-engage.log)
![liveevent-publications](https://user-images.githubusercontent.com/34450887/114988883-200e7200-9e97-11eb-8cf5-b23fc44092e1.jpg)

Initial topic on users list: 
https://groups.google.com/a/opencast.org/g/users/c/Zc_R42qBCcw

This is my first  bug report, please let me know if I missed anything. I'll be happy to provide further information.
regards
Christoph

**Comment by rute-santos** [comment]
I spent some time with Ruth and Christoph debugging this and here’s what we found:

In the log there are several of:
(LiveScheduleServiceImpl:295) - Creating live media package 11e18da5-1936-4e0b-a6e3-1570f88d7c14
which indicates that the LiveScheduleService never found the media package id in the search index (see code in [1]).

The LiveSchedulerService listens to Asset Manager update messages. It takes a snapshot when it first creates a "live" media package in the engage search index because it adds a publication element to it. When updating, no snapshots are taken (and the snapshot taken by itself at creation is ignored). The problem seems to be that, since it never finds the media package in the search index, it always "creates" one and takes a snapshot, causing the loop (there’s no check to ignore snapshot messages when creating).

It seems an ACL issue:

1) The media packages are published with the ACL: [ROLE_USER_ADMIN is allowed to read, ROLE_EXTERNAL_APPLICATION is allowed to read, ROLE_EXTERNAL_APPLICATION is allowed to write, ROLE_USER_ADMIN is allowed to write]

2) The LiveSchedulerService runs on the admin node and, when checking if a live mp exists, it uses this query:

    SearchQuery query = new SearchQuery().withId(mediaPackageId);
    SearchResult result = searchService.getByQuery(query); (*)

This query goes via the SearchServiceRemote:

    HttpGet get = new HttpGet(getSearchUrl(q, false));
    HttpResponse response = getResponse(get);

"false" indicates that the search is NOT going to use "getForAdministrativeRead"; it will use the current user's credentials.

3) When using a remote service, the opencast digest user is the one in context. In our local Opencast 5.x, this user has ROLE_SUDO and ROLE_ADMIN, but not ROLE_USER_ADMIN.

4) So it could be that the opencast digest user is not allowed to read that mp and nothing is returned in (*) so the LiveSchedulerServices creates the mp, takes a snapshot, gets a notification, creates it again, takes a snapshot, gets a notification, loop...

Ruth reported that the loop only happens when updating series metadata. If the episode metadata is updated, no looping happens.

Ruth then did the following test:

1) Create a series with the following ACLs (read and write), ROLE_USER_LANGR, ROLE_EXTERNAL_APPLICATION, ROLE_ADMIN
2) Add a scheduled live event —> success
3) Update some metadata of the episode —> success
4) Change the series title —> SUCCESS !!!

Note the additional ROLE_ADMIN when publishing.

Its still not clear to me why updating the episode metadata works, but I believe the fix for the series metadata update loop is: in the LiveSchedulerService, change the current user to the organization ‘admin’ and querying the engage search index with ‘getForAdministrativeRead’. Also, it doesn’t hurt to check the snapshot cache at create time in addition to at update time…

I will create a PR and ask them to test.
